src/Storage/Device/S3.php

```diff
@@ -6,6 +6,30 @@
 
 class S3 extends Device
 {
+    protected $accessKey;
+    protected $secretKey;
+    protected $bucket;
+    protected $region;
+    protected $acl='private';
+    protected $root = 'temp';
+    private $headers = array(
+		'Host' => '', 'Date' => '', 'Content-MD5' => '', 'Content-Type' => ''
+	);
+    private $response;
+    private $amzHeaders;
+
+    public function __construct($root='', $accessKey,$secretKey, $bucket, $region, $acl) {
```

Comment:
default value for the first key don't do anything as you have to pass it anyway.